Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1525,6 +1525,47 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
+
+    if let modifiers = node.modifiers, modifiers.hasError {
+      for modifier in modifiers {
+        guard let detail = modifier.detail else {
+          continue
+        }
+
+        let unexpectedTokens: [TokenSyntax] = [detail.unexpectedBetweenLeftParenAndDetail, detail.unexpectedBetweenDetailAndRightParen]
+          .compactMap { $0?.tokens(viewMode: .all) }
+          .flatMap { $0 }
+
+        // If there is no unexpected tokens it means we miss a paren or set keyword.
+        // So we just skip the handling here
+        guard let firstUnexpected = unexpectedTokens.first else {
+          continue
+        }
+
+        let fixItMessage: ParserFixIt
+
+        if unexpectedTokens.count == 1 {
```

Comment:
How is whether we replace or remove tokens related to the number of unexpected tokens. Shouldn’t this be the following?

```suggestion
        if detail.detail.presence == .missing {
```

And that got me thinking. If `detail.detail` is present, we shouldn’t be emitting a `MissingNodesError` at all. In that case we should probably just `continue` here and emit a defaulted `unexpected code` diagnostic.

Reply:
It's because if we have `private(get)` I want to replace it with `(set)`. 
Or should we then remove get and the left/right paren.

Reply:
https://github.com/apple/swift-syntax/pull/1643/files#r1190729627

Reply:
https://github.com/apple/swift-syntax/pull/1643#discussion_r1190729627

Reply:
I think this raises the general question of whether we should create a Fix-It to remove unexpected tokens. So far I have not done so because my thinking was that most likely the user wrote this code for a reason and most likely the solution isn’t to just delete it. If you took that view to the extreme, for every function that has a compilation error, you could offer to just delete the function body. 

So: I think it’s perfectly consistent to produce a ` unexpected code 'get,' in modifier` diagnostic without a Fix-It. 

But I’m open to have a discussion about this decision, so far it’s just been what I thought best and my thoughts above are the only opinion that has influenced the decision so far.

Reply:
I see a downside, that the missing `set` will automatically have a `Insert 'set'` Fix-It. 
This will result `private(get)` becomes `private(setget)`. Not sure what one is not better as we then produce more invalid code

Reply:
Does it though? In case `set` is missing, we will still enter this branch because `detail.detail.presence == .missing` is `true`. Also, in the `private(get)` test case, the Fix-It is `replace 'get' with 'set'` and the fixed source is `private(set)`.

Reply:
Fixed it now. 
Can you please review again @ahoppen ?